open_event/static/js/jquery/jquery.codezero.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * Created by Niranjan on 29-May-16.
+ */
+
+/**
+ * An extended jQuery method to call a callback on an input change
+ */
+jQuery.fn.extend({
+    valueChange: function (callback) {
+        return this.each(function () {
+            var elem = $(this);
+            elem.data('oldVal', elem.val());
+            elem.bind("propertychange change click keyup input paste", function (event) {
+                if (elem.data('oldVal') != elem.val()) {
+                    elem.data('oldVal', elem.val());
+                    callback(elem.val(), event);
+                }
+            });
+        });
+    }
+});
+
+/**
+ * Cache the results of a RegExp match.
+ * @type {Function}
+ */
+var cachedFuzzyMatch = _.memoize(function (pattern) {
+    return new RegExp(pattern.split("").reduce(function (a, b) {
+        return a + '[^' + b + ']*' + b;
+    }), 'i');
+});
+
+/**
+ * Fuzzy match a string and a pattern/query
+ * @param {string} str The target string
+ * @param {string} pattern The query string
+ * @returns {boolean} true if matches. Else false.
+ */
+function fuzzyMatch(str, pattern) {
+    return cachedFuzzyMatch(pattern).test(str)
+}
+
+/**
+ * Round of the the closest multiple of a number specified
+ * @param {int} val The number to round off
+ * @param {int} [multiple=6] The number to whose closest multiple val has to be rounded off to
+ * @returns {int}
+ */
+function roundOffToMultiple(val, multiple) {
+    if (!multiple) {
+        multiple = 6;
+    }
+    if (val > 0) {
+        var roundUp = Math.ceil(val / multiple) * multiple;
+        var roundDown = Math.floor(val / multiple) * multiple;
+
+        if (val >= roundDown + 12) {
+            return roundUp;
+        } else {
+            return roundDown;
+        }
+
+    } else {
+        return multiple;
+    }
+}
+
+/**
+ * Check if an element is horizontally bound within a parent div
+ * @param {jQuery} $parentDiv The parent div
+ * @param {jQuery} $childDiv The child div target
+ * @param {int} [offsetAdd=0] Offset to add while checking
+ * @returns {boolean}
+ */
+function horizontallyBound($parentDiv, $childDiv, offsetAdd) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: ['horizontallyBound' is defined but never used.](https://www.codacy.com/app/triplez/open-event-orga-server/file/2674855960/issues/source?bid=3233864&fileBranchId=3324174#l75)